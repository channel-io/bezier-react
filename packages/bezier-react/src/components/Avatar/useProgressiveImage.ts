import {
  useEffect,
  useState,
} from 'react'

enum ImageEventType {
  Load = 'load',
  Error = 'error',
}

export interface CachedImage {
  src: string
  isLoaded: boolean
}

type ImageCacheMap = Map<string, CachedImage>

const defaultImageCache = new Map<string, CachedImage>()

function getCachedImage(src: string, imageCache: ImageCacheMap) {
  const cachedImage = imageCache.get(src)
  if (!cachedImage) { return null }
  return cachedImage
}

export default function useProgressiveImage(
  src: string,
  defaultSrc: string,
  imageCache: ImageCacheMap = defaultImageCache,
) {
  const [source, setSource] = useState<CachedImage | null>(() => getCachedImage(src, imageCache))

  useEffect(function updateSource() {
    if (source?.src === src) { return undefined }

    const cachedImage = getCachedImage(src, imageCache)

    if (cachedImage?.isLoaded) {
      setSource(cachedImage)
      return undefined
    }

    const image = new Image()
    image.src = src

    function loadImage(event: Event) {
      const loadedImage = {
        src,
        isLoaded: event.type === ImageEventType.Load,
      }
      setSource(loadedImage)
      imageCache.set(src, loadedImage)
    }

    image.addEventListener(ImageEventType.Load, loadImage)
    image.addEventListener(ImageEventType.Error, loadImage)

    return function cleanup() {
      image.removeEventListener(ImageEventType.Load, loadImage)
      image.removeEventListener(ImageEventType.Error, loadImage)
    }
  }, [
    src,
    source,
    imageCache,
  ])

  if (!source || !source.isLoaded) {
    return defaultSrc
  }

  return source.src
}

