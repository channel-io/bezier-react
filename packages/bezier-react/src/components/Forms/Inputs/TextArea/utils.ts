import { type SemanticNames } from '~/src/foundation'

interface CalculateBgColorProps {
  focused: boolean
  hasError: boolean
  readOnly: boolean
}

export function getTextAreaBgColorSemanticName({
  focused,
  hasError,
  readOnly,
}: CalculateBgColorProps): SemanticNames {
  if (readOnly) { return 'bg-grey-lighter' }
  if (focused || hasError) { return 'bg-white-normal' }
  return 'bg-grey-lightest'
}
