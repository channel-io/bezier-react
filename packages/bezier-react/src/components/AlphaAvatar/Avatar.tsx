'use client'

import React, { forwardRef, useMemo } from 'react'

import classNames from 'classnames'

import { isEmpty } from '~/src/utils/type'

import { useAvatarContext } from '~/src/components/AlphaAvatar/AvatarSizeContext'
import {
  SmoothCornersBox,
  type SmoothCornersBoxProps,
} from '~/src/components/SmoothCornersBox'
import { Status, type StatusSize } from '~/src/components/Status'
import { useTokens } from '~/src/components/ThemeProvider'

import type { AvatarProps, AvatarSize } from './Avatar.types'
import defaultAvatarUrl from './assets/default-avatar.svg'
import useProgressiveImage from './useProgressiveImage'

import styles from './Avatar.module.scss'

function getStatusSize(size: AvatarSize): StatusSize {
  switch (size) {
    case '90':
    case '120':
      return 'l'
    default:
      return 'm'
  }
}

function getShadow(size: AvatarSize): SmoothCornersBoxProps['shadow'] {
  const spreadRadius = (() => {
    switch (size) {
      case '90':
        return 3
      case '120':
        return 4
      default:
        return 2
    }
  })()

  return {
    spreadRadius,
    color: 'bg-white-high',
  }
}

export function useAvatarRadiusToken() {
  // TODO: Replace this with useAlphaTokens
  return useTokens().global.radius['radius-42-p']
}

export const AVATAR_WRAPPER_TEST_ID = 'bezier-avatar-wrapper'
export const AVATAR_TEST_ID = 'bezier-avatar'
export const STATUS_WRAPPER_TEST_ID = 'bezier-status-wrapper'

/**
 * `Avatar` is a component for representing some profile image.
 * @example
 *
 * ```tsx
 * <Avatar
 *   avatarUrl="https://..."
 *   name="channel"
 *   size="48"
 *   showBorder
 *   disabled
 * />
 * ```
 */
export const Avatar = forwardRef<HTMLDivElement, AvatarProps>(function Avatar(
  {
    avatarUrl = '',
    fallbackUrl = defaultAvatarUrl,
    size: sizeProps = '24',
    name,
    disabled = false,
    showBorder = false,
    smoothCorners = true,
    status,
    className,
    children,
    ...rest
  },
  forwardedRef
) {
  const avatarGroupContext = useAvatarContext()
  const size = avatarGroupContext?.size ?? sizeProps
  const loadedAvatarUrl = useProgressiveImage(avatarUrl, fallbackUrl)
  const AVATAR_BORDER_RADIUS = useAvatarRadiusToken()

  const StatusComponent = useMemo(() => {
    if (
      (isEmpty(children) && !status) ||
      (children && !React.isValidElement(children))
    ) {
      return null
    }

    const Contents = (() => {
      if (children) {
        return children
      }
      if (status) {
        return (
          <Status
            type={status}
            size={getStatusSize(size)}
          />
        )
      }
      return null
    })()

    return (
      Contents && (
        <div
          className={styles.StatusWrapper}
          data-testid={STATUS_WRAPPER_TEST_ID}
        >
          {Contents}
        </div>
      )
    )
  }, [status, size, children])

  return (
    <div
      className={classNames(
        styles.Avatar,
        styles[`size-${size}`],
        disabled && styles.disabled,
        className
      )}
      data-disabled={disabled}
      data-testid={AVATAR_WRAPPER_TEST_ID}
      {...rest}
    >
      <SmoothCornersBox
        ref={forwardedRef}
        aria-description={name}
        className={classNames(
          styles.AvatarImage,
          Number(size) >= 72 && styles['big-size'],
          showBorder && styles.bordered
        )}
        disabled={!smoothCorners}
        borderRadius={AVATAR_BORDER_RADIUS}
        shadow={showBorder ? getShadow(size) : undefined}
        backgroundColor="bg-white-normal"
        backgroundImage={loadedAvatarUrl}
        data-testid={AVATAR_TEST_ID}
      >
        {StatusComponent}
      </SmoothCornersBox>
    </div>
  )
})
