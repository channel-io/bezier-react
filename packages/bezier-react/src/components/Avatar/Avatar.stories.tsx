import React from 'react'

import { type Meta, type StoryFn, type StoryObj } from '@storybook/react'

import { Avatar } from './Avatar'
import type { AvatarProps } from './Avatar.types'

const MOCK_AVATAR_URL =
  'https://cf.channel.io/thumb/200x200/pub-file/1/65fc43ee585607b276f6/tmp-3329819395'

const meta: Meta<typeof Avatar> = {
  component: Avatar,
  argTypes: {
    onClick: {
      action: 'clicked',
    },
    onMouseEnter: {
      action: 'mouseEnter',
    },
    onMouseLeave: {
      action: 'mouseLeave',
    },
  },
}
export default meta

const Template: StoryFn<AvatarProps> = (args) => <Avatar {...args} />

export const Primary: StoryObj<AvatarProps> = {
  render: Template,

  args: {
    avatarUrl: MOCK_AVATAR_URL,
    name: 'Channel',
    size: '24',
    showBorder: false,
    disabled: false,
    smoothCorners: true,
  },
}

const TemplateWithCustomStatus: StoryFn<AvatarProps> = (args) => (
  <Avatar {...args}>
    <Avatar
      avatarUrl="https://bit.ly/kent-c-dodds"
      name="Kent Dodds"
      size="20"
      showBorder
    />
  </Avatar>
)

export const WithCustomStatus: StoryObj<AvatarProps> = {
  render: TemplateWithCustomStatus,

  args: {
    avatarUrl: MOCK_AVATAR_URL,
    name: 'Channel',
    size: '48',
    showBorder: false,
    disabled: false,
  },
}
