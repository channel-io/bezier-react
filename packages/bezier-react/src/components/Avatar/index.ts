import {
  Avatar,
  useAvatarRadiusToken,
} from './Avatar'
import { AvatarSize } from './Avatar.types'
import type AvatarProps from './Avatar.types'

export type {
  AvatarProps,
}

export {
  Avatar,
  useAvatarRadiusToken,
  AvatarSize,
}
