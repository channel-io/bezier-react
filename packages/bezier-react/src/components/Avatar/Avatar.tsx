import React, {
  forwardRef,
  useMemo,
} from 'react'

import classNames from 'classnames'

import { isEmpty } from '~/src/utils/type'

import {
  AlphaSmoothCornersBox,
  type AlphaSmoothCornersBoxProps,
} from '~/src/components/AlphaSmoothCornersBox'
import {
  Status,
  StatusSize,
} from '~/src/components/Status'
import { useToken } from '~/src/components/ThemeProvider'

import type { AvatarProps } from './Avatar.types'
import { AvatarSize } from './Avatar.types'
import defaultAvatarUrl from './assets/default-avatar.svg'
import useProgressiveImage from './useProgressiveImage'

import styles from './Avatar.module.scss'

const shadow: AlphaSmoothCornersBoxProps['shadow'] = {
  spreadRadius: 2,
  color: 'bg-white-high',
}

export function useAvatarRadiusToken() {
  return useToken().global.radius['radius-42-p']
}

export const AVATAR_WRAPPER_TEST_ID = 'bezier-react-avatar-wrapper'
export const AVATAR_TEST_ID = 'bezier-react-avatar'
export const STATUS_WRAPPER_TEST_ID = 'bezier-react-status-wrapper'

/**
 * `Avatar` is a component for representing some profile image.
 * @example
 *
 * ```tsx
 * <Avatar
 *   avatarUrl="'https://cf.channel.io/thumb/200x200/pub-file/1/606d87d059a6093594c0/ch-symbol-filled-smiley-bg.png"
 *   name="channel"
 *   size={AvatarSize.Size48}
 *   showBorder
 *   disabled
 * />
 * ```
 */
export const Avatar = forwardRef<HTMLDivElement, AvatarProps>(function Avatar({
  avatarUrl = '',
  fallbackUrl = defaultAvatarUrl,
  size = AvatarSize.Size24,
  name,
  testId = AVATAR_TEST_ID,
  disabled = false,
  showBorder = false,
  smoothCorners = true,
  status,
  className,
  children,
  ...rest
}, forwardedRef) {
  const loadedAvatarUrl = useProgressiveImage(avatarUrl, fallbackUrl)
  const AVATAR_BORDER_RADIUS = useAvatarRadiusToken()

  const StatusComponent = useMemo(() => {
    if (
      (isEmpty(children) && !status)
      || (children && !React.isValidElement(children))
    ) {
      return null
    }

    const statusSize = size >= AvatarSize.Size90 ? StatusSize.L : StatusSize.M

    const Contents = (() => {
      if (children) { return children }
      if (status) {
        return (
          <Status
            type={status}
            size={statusSize}
          />
        )
      }
      return null
    })()

    return Contents && (
      <div
        data-testid={STATUS_WRAPPER_TEST_ID}
        className={styles.StatusWrapper}
      >
        { Contents }
      </div>
    )
  }, [
    status,
    size,
    children,
  ])

  return (
    <div
      data-testid={AVATAR_WRAPPER_TEST_ID}
      data-disabled={disabled}
      className={classNames(
        styles.Avatar,
        styles[`size-${size}`],
        disabled && styles.disabled,
        className,
      )}
      {...rest}
    >
      <AlphaSmoothCornersBox
        ref={forwardedRef}
        testId={testId}
        aria-label={name}
        className={classNames(
          styles.AvatarImage,
          size >= AvatarSize.Size72 && styles['big-size'],
          showBorder && styles.bordered,
        )}
        disabled={!smoothCorners}
        borderRadius={AVATAR_BORDER_RADIUS}
        shadow={showBorder ? shadow : undefined}
        backgroundColor="bg-white-normal"
        backgroundImage={loadedAvatarUrl}
      >
        { StatusComponent }
      </AlphaSmoothCornersBox>
    </div>
  )
})
