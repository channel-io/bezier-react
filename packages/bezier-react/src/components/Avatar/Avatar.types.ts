import type {
  BezierComponentProps,
  ChildrenProps,
  DisableProps,
  SizeProps,
} from '~/src/types/props'

import { type StatusType } from '~/src/components/Status'

export type AvatarSize =
| '20'
| '24'
| '30'
| '36'
| '42'
| '48'
| '72'
| '90'
| '120'

interface AvatarOwnProps {
  /**
   * Semantic name of the avatar.
   */
  name: string
  /**
   * Asset image URL of the avatar.
   * @default ''
   */
  avatarUrl?: string
  /**
   * Fallback image URL for the avatar.
   * @default DEFAULT_AVATAR_URL
   */
  fallbackUrl?: string
  /**
   * Additional status type for this avatar.
   * Typically, this is used to indicate the status of the user (online, offline, etc).
   *
   * If additional customization of the status component is needed,
   * pass the custom status component as `children` of the Avatar component.
   */
  status?: StatusType
  /**
   * Whether to display the border of the avatar.
   * @default false
   */
  showBorder?: boolean
  /**
   * Whether to use the smooth corners feature.
   * Overrides whether the smooth corners feature is applied when it is enabled.
   * @default true
   */
  smoothCorners?: boolean
}

export interface AvatarProps extends
  BezierComponentProps<'div'>,
  SizeProps<AvatarSize>,
  DisableProps,
  ChildrenProps,
  AvatarOwnProps {}
