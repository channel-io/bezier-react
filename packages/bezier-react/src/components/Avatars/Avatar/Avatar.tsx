import React, {
  forwardRef,
  useMemo,
} from 'react'

import classNames from 'classnames'

import { isEmpty } from '~/src/utils/typeUtils'

import { type BoxShadow } from '~/src/components/AlphaSmoothCornersBox'
import { AVATAR_BORDER_RADIUS_PERCENTAGE } from '~/src/components/Avatars/AvatarStyle'
import {
  Status,
  StatusSize,
} from '~/src/components/Status'

// NOTE: Don't fix it. When using absolute paths, a module resolve error occurs at the runtime of the library consumer.
// eslint-disable-next-line no-restricted-imports
import defaultAvatarUrl from '../assets/defaultAvatar.svg'

import type AvatarProps from './Avatar.types'
import { AvatarSize } from './Avatar.types'
import useProgressiveImage from './useProgressiveImage'

import * as Styled from './Avatar.styled'

const shadow: BoxShadow = {
  spreadRadius: 2,
  color: 'bg-white-high',
}

export const AVATAR_WRAPPER_TEST_ID = 'bezier-react-avatar-wrapper'
export const AVATAR_TEST_ID = 'bezier-react-avatar'
export const STATUS_WRAPPER_TEST_ID = 'bezier-react-status-wrapper'

export const Avatar = forwardRef(function Avatar({
  avatarUrl = '',
  fallbackUrl = defaultAvatarUrl,
  size = AvatarSize.Size24,
  name,
  testId = AVATAR_TEST_ID,
  disabled = false,
  showBorder = false,
  smoothCorners = true,
  status,
  className: classNameProp,
  wrapperClassName,
  interpolation,
  wrapperInterpolation,
  style: styleProp,
  children,
  ...rest
}: AvatarProps, forwardedRef: React.Ref<HTMLDivElement>) {
  const loadedAvatarUrl = useProgressiveImage(avatarUrl, fallbackUrl)

  const StatusComponent = useMemo(() => {
    if (
      (isEmpty(children) && !status)
      || (children && !React.isValidElement(children))
    ) {
      return null
    }

    const statusSize = size >= AvatarSize.Size90 ? StatusSize.L : StatusSize.M

    const Contents = (() => {
      if (children) { return children }
      if (status) {
        return (
          <Status
            type={status}
            size={statusSize}
          />
        )
      }
      return null
    })()

    return (
      <Styled.StatusWrapper data-testid={STATUS_WRAPPER_TEST_ID}>
        { Contents }
      </Styled.StatusWrapper>
    )
  }, [
    status,
    size,
    children,
  ])

  const style = useMemo(() => ({
    ...styleProp,
    '--bezier-avatar-status-gap': `${size >= AvatarSize.Size72 ? 4 : -2}px`,
  }), [
    styleProp,
    size,
  ])

  const className = classNames(
    classNameProp,
    size && `size-${size}`,
    showBorder && 'bordered',
  )

  return (
    <Styled.AvatarWrapper
      data-testid={AVATAR_WRAPPER_TEST_ID}
      data-disabled={disabled}
      className={wrapperClassName}
      interpolation={wrapperInterpolation}
    >
      <Styled.AvatarImage
        {...rest}
        ref={forwardedRef}
        data-testid={testId}
        aria-label={name}
        style={style}
        className={className}
        interpolation={interpolation}
        disabled={!smoothCorners}
        borderRadius={`${AVATAR_BORDER_RADIUS_PERCENTAGE}%`}
        shadow={showBorder ? shadow : undefined}
        backgroundColor="bg-white-normal"
        backgroundImage={loadedAvatarUrl}
      >
        { StatusComponent }
      </Styled.AvatarImage>
    </Styled.AvatarWrapper>
  )
})
