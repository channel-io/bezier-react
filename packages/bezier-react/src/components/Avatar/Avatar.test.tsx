import React from 'react'

import { render } from '~/src/utils/test'

import { StatusType } from '~/src/components/Status'

import {
  AVATAR_TEST_ID,
  AVATAR_WRAPPER_TEST_ID,
  Avatar,
  STATUS_WRAPPER_TEST_ID,
} from './Avatar'
import type { AvatarProps } from './Avatar.types'
import { AvatarSize } from './Avatar.types'

describe('Avatar >', () => {
  let props: AvatarProps

  const mockAvatarUrl = 'https://bit.ly/dan-abramov'
  const mockFallbackUrl = 'https://www.google.com'

  beforeEach(() => {
    props = {
      avatarUrl: mockAvatarUrl,
      fallbackUrl: mockFallbackUrl,
      name: 'Name',
    }
  })

  afterAll(() => {
    jest.restoreAllMocks()
  })

  // NOTE: unavailable smoothCorners
  const renderAvatar = (optionProps?: Partial<AvatarProps>) => render(
    <Avatar {...props} {...optionProps} />,
  )

  it('Snapshot', () => {
    const { getByTestId } = renderAvatar()
    const img = getByTestId(AVATAR_TEST_ID)

    expect(img).toMatchSnapshot()
  })

  it('renders disabled style', () => {
    const { getByTestId } = renderAvatar({ disabled: true })
    const wrapper = getByTestId(AVATAR_WRAPPER_TEST_ID)

    expect(wrapper).toHaveStyle('opacity: var(--opacity-disabled)')
  })

  it('renders border style', () => {
    const { getByTestId } = renderAvatar({ showBorder: true })
    const img = getByTestId(AVATAR_TEST_ID)

    // NOTE(@ed): psuedo element 스타일 테스트가 어려워, 스냅샷 테스트로 대체
    expect(img).toMatchSnapshot()
  })

  it('should have right -2px, bottom -2px style on StatusWrapper', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })

  it('should have right 2px, bottom 2px style on StatusWrapper when show border', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online, showBorder: true })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })

  it('should have right 4px, bottom 4px style on StatusWrapper when size grater then 72', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online, size: AvatarSize.Size72 })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })

  it('should have right 8px, bottom 8px style on StatusWrapper when size grater then 72 and show border', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online, size: AvatarSize.Size72, showBorder: true })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })

  it('should have right 4px, bottom 4px style on StatusWrapper when size grater then 90', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online, size: AvatarSize.Size90 })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })

  it('should have right 8px, bottom 8px style on StatusWrapper when size grater then 90 and show border', () => {
    const { getByTestId } = renderAvatar({ status: StatusType.Online, size: AvatarSize.Size90, showBorder: true })
    const statusWrapper = getByTestId(STATUS_WRAPPER_TEST_ID)

    expect(statusWrapper).toMatchSnapshot()
  })
})
