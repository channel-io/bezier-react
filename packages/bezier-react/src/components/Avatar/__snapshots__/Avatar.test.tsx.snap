// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar > Snapshot 1`] = `
<div
  aria-label="Name"
  class="SmoothCornersBox AvatarImage"
  data-state="disabled"
  data-testid="bezier-react-avatar"
  style="--b-smooth-corners-box-border-radius: 42%; --b-smooth-corners-box-shadow-blur-radius: 0px; --b-smooth-corners-box-shadow-spread-radius: 0px; --b-smooth-corners-box-padding: 0px; --b-smooth-corners-box-margin: 0px; --b-smooth-corners-box-background-color: var(--bg-white-normal); --b-smooth-corners-box-background-image: url(https://www.google.com);"
/>
`;

exports[`Avatar > renders border style 1`] = `
<div
  aria-label="Name"
  class="SmoothCornersBox AvatarImage bordered"
  data-state="disabled"
  data-testid="bezier-react-avatar"
  style="--b-smooth-corners-box-border-radius: 42%; --b-smooth-corners-box-shadow-blur-radius: 0px; --b-smooth-corners-box-shadow-spread-radius: 2px; --b-smooth-corners-box-shadow-color: var(--bg-white-high); --b-smooth-corners-box-padding: 4px; --b-smooth-corners-box-margin: 0px; --b-smooth-corners-box-background-color: var(--bg-white-normal); --b-smooth-corners-box-background-image: url(https://www.google.com);"
/>
`;

exports[`Avatar > should have right -2px, bottom -2px style on StatusWrapper 1`] = `
<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-m"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;

exports[`Avatar > should have right 2px, bottom 2px style on StatusWrapper when show border 1`] = `
<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-m"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;

exports[`Avatar > should have right 4px, bottom 4px style on StatusWrapper when size grater then 72 1`] = `
<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-m"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;

exports[`Avatar > should have right 4px, bottom 4px style on StatusWrapper when size grater then 90 1`] = `
<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-l"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;

exports[`Avatar > should have right 8px, bottom 8px style on StatusWrapper when size grater then 72 and show border 1`] = `
<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-m"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;

exports[`Avatar > should have right 8px, bottom 8px style on StatusWrapper when size grater then 90 and show border 1`] = `
<div
  class="StatusWrapper"
  data-testid="bezier-react-status-wrapper"
>
  <div
    class="Status size-l"
    style="--b-status-bg-color: var(--bgtxt-green-normal);"
  />
</div>
`;
